### ISSUE TITLE
the argument ref.lvl in sjmisc::to_factor() not working

### ISSUE THREAD

**Issue by frankcsliu** [strengejacke/sjmisc#133]
Hi, Daniel, I recently found that the argument is not effective any more and seems not recognizing labels. 
Example data to download [here](https://www.dropbox.com/s/wt2bbjvy53cfuff/teds2006_kao.rda?dl=0)

load(kao06)
library(sjmisc)
kao06$partyID <- rec(kao06$L2B,
                     rec="98:hi=NA; else=copy", 
                     val.labels = c("partyA","partyB","partyC",
                                    "partyD", "partyE", "others"), 
                     as.num=F)

# not working
kao06$partyID <- to_factor(kao06$partyID, ref.lvl="partyB")
kao06.mod.2 <- glm(turnout ~ partyID, family=binomial, data=kao06)
summary(kao06.mod.2)

# not working
kao06$partyID <- to_factor(kao06$partyID, ref.lvl="partyA")
summary(kao06.mod.2)

# not working
kao06$partyID <- to_factor(kao06$partyID, ref.lvl=2) 
summary(kao06.mod.2)

# this traditional method works: 
kao06$partyID <- ref_lvl(kao06$partyID, lvl=2)
summary(kao06.mod.2)

**Comment by strengejacke** [comment]
yes, `to_factor()` doesn't work, because your input is already a factor. That's why it's immediately returned, unchanged.

**Comment by frankcsliu** [comment]
Could you make it work for factor, too? I see it as a more sofiscated alternative to ref_lvl() because I like the argument of specifiying label. Thanks.
